### ISSUE TITLE
Add the Playoff Bracket

### ISSUE THREAD

**Issue by 1fish2** [the-blue-alliance/the-blue-alliance-android#686]
It'd be really nice to see the Playoff Bracket, esp. while elimination matches are happening.

**Comment by phil-lopreiato** [comment]
@nwalters512 what have you been thinking for this?

**Comment by nwalters512** [comment]
Ah, yeah, I gotta start working on this. Based on what I remember from our discussion, the "best way" of doing this would look something like

 - for the first level of matches, show the alliance # and team #s
    - should team numbers be displayed vertically, or on a single line in a comma-separated list?
 - for the remaining matches, show just the alliance number; show team numbers when the alliance number is tapped
 - orient the bracket horizontally, with the first level of matches aligned to the left

Things probably get kind of messy with octofinals though. If the team numbers for the first level are stacked vertically, you'll end up with a situation where you probably have to scroll vertically to see all the matches. If the team numbers are instead listed horizontally, you'll instead probably have to scroll horizontally to see the full bracket.

I'm more of a fan of the first situation; if the bracket has to scroll horizontally, it would make swiping through the tabs more difficult because the bracket would capture the swipe before the tab got it.

Any more thoughts on this?

**Comment by 1fish2** [comment]
Can you sketch those? A scan or photo of a quick pencil sketch would be fine, and much easier to visualize. I can also show it to a visual designer friend.

I agree about not scrolling horizontally so it doesn't capture swipes.

**Comment by nwalters512** [comment]
Here are the two options I proposed. You can see the tradeoffs between horizontal and vertical compactness.

![Imgur](http://i.imgur.com/7LpZTJN.jpg)

**Comment by 1fish2** [comment]
I vote for the proposal on the left (vertical list of team numbers).

This might require vertical scrolling but the diagram is continuous so there won't be information hidden unexpectedly "below the fold." It also makes it OK to add info like maybe the next match time on an alliance box or on a playoff box.

How about some way to show visually which input line flows through a playoff box to its output? The alliance number is self-explanatory but not glanceable.

Red/blue fill colors are ambiguous in the intermediate playoff boxes. If it shows the color of the winning alliance during those matches, then the diagram doesn't show what color that alliance has going into the next playoff box, and thus the fill color of that next box doesn't have a key. One solution would be to color the right side of the playoff box with the alliance's color going into the next playoff.

I'd also indicate the user's favorite teams, e.g. with bold team numbers.

**Comment by phil-lopreiato** [comment]
The other thing to consider is octofinals. 
a) Do we even care about displaying all 16 alliances?
b) It'll double the vertical space and add one more column

**Comment by nwalters512** [comment]
@1fish2 yeah, I figured alliance colors should behave like that; I just didn't really illustrate that well in my example.

We still need to figure out an app-wide way to indicate favorite models. I say we hold off on trying to do that here until we figure out what that universal indicator is going to be.

With regards to octofinals: I think it's probably best to at least attempt to display the full bracket, at least for consistency's sake. It may get a little dicey on narrow low dpi devices though. I can test that and if it doesn't work, we can re-evaluate. That being said, it won't add *that* much horizontal space, so it's probably doable.

**Comment by 1fish2** [comment]
Abutting the boxes in the first column saves a column:

![playoff bracket 1](https://cloud.githubusercontent.com/assets/1043120/14435201/7a589e86-ffcb-11e5-928a-7f9fda751c82.png)

**Comment by 1fish2** [comment]
FWIW, image search results for [playoff bracket] all look about like this or two-sided versions of this:

![league playoff bracket](https://cloud.githubusercontent.com/assets/1043120/14435443/cac8aedc-ffcc-11e5-8322-1b50a7b3924b.jpg)
